### PR TITLE
Basic readme describing what we're trying to achieve

### PR DESCRIPTION
**Pull request by talesh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,65 @@
...
+| Label           | Description                                                                                                                   |
+| --------------- | ----------------------------------------------------------------------------------------------------------------------------- |
+| **bugfix**      | The issue or pull request relates to bug fixing.                                                                              |
+| **enhancement** | The issue or pull request that raises the MFTF to a higher degree (for example new features, optimization, refactoring, etc). |
+
+
+
+## Reporting security issues
+
+To report security vulnerabilities and other security issues in the Magento software or web sites, send an email with the report at [security@magento.com](mailto:security@magento.com).
```
Comment (mpchadwick) [inline README.md]: Shouldn't people be using the bug bounty program?

**Comment by jonathanKingston** [comment]
I would personally start with linting first before fuzzing.

For example we built https://github.com/mozilla/eslint-plugin-no-unsanitized to find code that could be vulnerable to XSS attacks. The same could be used and modified for all JavaScript, similar could be built for SQL and PHP.
Designing around linters is certainly possible but the scope is certainly much smaller. An alternative would be reimagining modules in such a way that fuzzing becomes simpler.

As far as I understand the module system in Magento is it's too dynamic to figure out the entry points into the code, also what does a failure look like? Given the versatile nature of a module, it might be designed to drop all the rows in a table or restart the web server, how would a fuzzer determine what is insecure and what was designed?

File: README.md (hunk 1 above)
Comment (talesh) [reply README.md]: This is a copy/paste from the MFTF readme because I'm lazy. Maybe @maksek knows best.

**Comment by talesh** [comment]
Hi @jonathanKingston, 

I do believe that linting type functionality already exists in the [Magento Extension Quality Program Coding Standard](https://github.com/magento/marketplace-eqp). Probably that is the best place to add rules to block XSS attacks. 

I agree fuzzing in Magento is not simple and trying to build something to do so automatically is even more difficult. This project is still in ideas-as-backlog mode. Throw in your ideas on how you envision it as issues and tag as "Ideas" and I think together we'll find a path to minimal viability and then extend.

**Comment by jonathanKingston** [comment]
Ah I see.

Rather than code sniff for forbidden functions https://github.com/magento/marketplace-eqp/blob/master/MEQP1/Sniffs/Security/DiscouragedFunctionSniff.php compiling a version of PHP to crash when they are used would be a start, you likely could spin up a headless test to click to areas of Magento that appear to be used by the module. Similar for slow performing db queries making the connection crash (or similar clear failure that the fuzzer can catch).

> Throw in your ideas on how you envision it as issues and tag as "Ideas" and I think together we'll find a path to minimal viability and then extend.

I don't really know Magento at all tbh, I'm sure someone like @shakyshane or others at @wearejh might be interested in some form of automation testing though.

**Pull request merged**